test/suites/integration/instance_tag_test.go

```diff
@@ -0,0 +1,144 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration_test
+
+import (
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/test"
+	"github.com/aws/karpenter/pkg/apis/settings"
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+	awstest "github.com/aws/karpenter/pkg/test"
+)
+
+var _ = Describe("EC2 Instance Tagging", func() {
+	var nodeTemplateOpts v1alpha1.AWSNodeTemplateSpec
+	var provisionerOpts test.ProvisionerOptions
+
+	BeforeEach(func() {
+		nodeTemplateOpts = v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{
+			SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+			SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+		}}
+
+		provisionerOpts = test.ProvisionerOptions{
+			Requirements: []v1.NodeSelectorRequirement{
+				{
+					Key:      v1alpha5.LabelCapacityType,
+					Operator: v1.NodeSelectorOpIn,
+					Values:   []string{v1alpha1.CapacityTypeOnDemand},
+				},
+				{
+					Key:      v1.LabelOSStable,
+					Operator: v1.NodeSelectorOpIn,
+					Values:   []string{string(v1.Linux)},
+				},
+				{
+					Key:      "karpenter.k8s.aws/instance-hypervisor",
+					Operator: v1.NodeSelectorOpIn,
+					Values:   []string{"nitro"},
+				},
+			},
+			Limits: v1.ResourceList{},
+		}
+	})
+
+	It("should overwrite default Name tags", func() {
+		nodeTemplate := awstest.AWSNodeTemplate(nodeTemplateOpts)
+		provisioner := test.Provisioner(provisionerOpts, test.ProvisionerOptions{ProviderRef: &v1alpha5.MachineTemplateRef{
+			Name: nodeTemplate.Name,
+		}})
+		pod := test.Pod()
+
+		env.ExpectCreated(provisioner, nodeTemplate, pod)
+		env.EventuallyExpectCreatedMachineCount("==", 1)
+		env.EventuallyExpectCreatedNodeCount("==", 1)
+		node := env.EventuallyExpectInitializedNodeCount("==", 1)[0]
+		nodeName := client.ObjectKeyFromObject(node)
+
+		Eventually(func(g Gomega) {
+			node = &v1.Node{}
+			lo.Must0(env.Client.Get(env.Context, nodeName, node))
+			g.Expect(node.Annotations["karpenter.sh/instance-tagged"]).To(Equal("true"))
+		}, 30*time.Second)
+
+		resourceID := lo.Must(getResourceID(node.Spec.ProviderID))
+		tags := getResourceTags(resourceID)
+		Expect(tags["Name"]).To(Equal(node.Name))
+	})
+
+	It("shouldn't overwrite custom Name tags", func() {
```

Comment:
Can we add a test case for validating that we don't tag instances that are provisioned from Provisioners/Machines?